## Final Project Report
#### Sam Du
#### EE282 Fall 2020

___

### Introduction

For the rest of the introduction, please refer to the topic and analysis proposals. 

The double-knockout mice (DKO) in this study were compared at four different time points: non-bleached (NB), 6 hours, 1 day, and 3 days post-bleach (6h, 1d, and 3d respectively). After sacrifice, the eye was enucleated and an eyecup was created, at which point RPE was isolated from the eyecup and prepared for library preparation. Four mice were analyzed at each timepoint.

___

### Methods

#### Reference Genome

The _Mus musculus_ reference genome was downloaded from ENSEMBL. I used the GRCm38 release of the top level genome. I also downloaded the GTF for this genome. Checksums were verified against the values generated by `sum` and the ENSEMBL provided checksums.

```bash
wget ftp://ftp.ensembl.org/pub/release-102/fasta/mus_musculus/dna/Mus_musculus.GRCm38.dna.toplevel.fa.gz
wget ftp://ftp.ensembl.org/pub/release-102/gtf/mus_musculus/Mus_musculus.GRCm38.102.gtf.gz 
```

I indexed the genome with BWA index (v0.7.8-r455).

I also downloaded the ENSEMBL gene annotations from Biomart for GRCm38 that reports gene name, type, description, and ENSEMBL ID and created a file named `ENSEMBL_annotations.txt` in my `references` folder.

#### Data Acquisition

Data was downloaded through the European Nucleotide Archive. FTP links for runs SRR12148400 through SRR12148415 were added into a file `download.txt` located in `/data/rawdata/`. Files were batch downloaded with wget with the following command:

```bash
wget -i download.txt
```

#### Data Pre-Processing

FASTQC was used to analyze read quality. The shell scripts [`fastqc.sh`](https://github.com/swd12012/ee282/blob/finalProject/project/scripts/fastqc.sub) and [`fastqc_dir.sh`](https://github.com/swd12012/ee282/blob/finalProject/project/scripts/fastqc_dir.sh) in the directory `/scripts/` were used to run FASTQC on all the `*.gz` files in my specified directory and output the FastQC files in the same directory:

```bash
sh /data/homezvol2/swdu/ee282/project/scripts/fastqc_dir.sh /data/homezvol2/swdu/ee282/project/data/rawdata/ /data/homezvol2/swdu/ee282/project/data/rawdata/
```

I created a [targets file](https://github.com/swd12012/ee282/blob/finalProject/project/data/targets.txt) which annotated my .bam files with sample name, group, condition, etc.

#### Data Alignment

Data was aligned with `bwa mem` and submitted to SLURM as a batch job with the two following scripts ([1](https://github.com/swd12012/ee282/blob/finalProject/project/scripts/bwamem.sub) [2](https://github.com/swd12012/ee282/blob/finalProject/project/scripts/bwamem10-15.sub)). The SAM files were then converted to BAM files with `samtools` (eg. `samtools view -bS *.sam > *.bam`).

#### Read Counting

I used the function `featureCounts` from the package Subread (v2.0.1) to perform my counts.

```bash
featureCounts -T 16 \
  -a /data/class/ecoevo282/swdu/ee282/project/reference/Mus_musculus.GRCm38.102.gtf.gz \
  -o /data/class/ecoevo282/swdu/ee282/project/data/processed/readCounts.txt \
  /data/class/ecoevo282/swdu/ee282/project/data/processed/*.bam
  ```

#### Data Visualization and PCA plotting

I used DESeq2 (v1.28.1) to create group- and sample-wise PCA plotting by variance stablizing transformation. This will help analyse the variance of each sample and each group and point to potential outliers that may need to be removed in following steps.

First, I cut out the columns that contain only the samples by running the following:

```bash
cat readcounts.txt \
| cut -f7-22 \
> readcounts_samplesonly.txt
```

I then created PCA plots for the dataset in R in the terminal:
```R
library(DESeq2)

#Read in count file as a dataframe
count_dataframe <- as.matrix(read.delim('readcounts_samplesonly.txt'), header=TRUE)

#Reorder count file by column
count_dataframe <- count_dataframe[,c(1,8:16, 2:7)]

#Read in targets file for sample annotation
targets <- read.delim('../targets', comment.char='#', header=TRUE, sep='\t')

#Create dataframes for PCA analysis, one based on sample and one based on group
colDataGroup <- data.frame(row.names=targets$SampleLong, conditions=targets$Factor)
colDataSample <- data.frame(row.names=targets$SampleLong, conditions=targets$SampleLong)

#Create DESeq dataset
sample_dataset <- DESeqDataSetFromMatrix(countData=count_dataframe, colData=colDataSample, design = ~ conditions)
group_dataset <- DESeqDataSetFromMatrix(countData=count_dataframe, colData=colDataGroup, design = ~ conditions)

#Apply variance stabilizing transformation
sample_vsd <- varianceStabilizingTransformation(sample_dataset)
group_vsd <- varianceStabilizingTransformation(group_dataset)

#Apply regularized log transformation
sample_rlog <- rlog(sample_dataset)
group_rlog <- rlog(group_dataset)

#Plot graphs
png('../../figures/sample_vsd_PCA.png', width=8, height=8, res=300, units='in')
plotPCA(sample_vsd, 'conditions')
dev.off()

png('../../figures/group_vsd_PCA.png', width=8, height=8, res=300, units='in')
plotPCA(group_vsd, 'conditions')
dev.off()

png('../../figures/sample_rlog_PCA.png', width=8, height=8, res=300, units='in')
plotPCA(sample_rlog, 'conditions')
dev.off()

png('../../figures/group_rlog_PCA.png', width=8, height=8, res=300, units='in')
plotPCA(group_rlog, 'conditions')
dev.off()
```

##### Differentially Expressed Gene (DEG) Analysis

DEG analysis was conducted with `edgeR`(v3.30.3) and `systemPipeR` (v1.24.2); both are R packages downloaded via Bioconductor. R version was 3.6.2 to accomodate `systemPipeR`.

```R
library(edgeR)
library(systemPipeR)

#Read in raw counts file and targets file

targets <- as.data.frame(read.delim('../targets.txt', header=T, comment.char='#'))
countDF <- as.data.frame(read.table('readcounts_sampleENSEMBL.txt', header=T, row.names=1))

#Define comparisons to be made, which are stated in the first line of the targets file
cmp <- readComp(file='../targets.txt', format='matrix', delim='-')

#Run edgeR
edgeDF <- run_edgeR(countDF=countDF, targets=targets, cmp=cmp[[1]], independent=TRUE, mdsplot="")

#Read in ENSEMBL annotations and remove first column (redundant information)
annotations <- read.delim('../../reference/ENSEMBL_annotations.csv', header=T, sep=',', row.names=1)
annotations <- annotations[,2-4]

#Merge annotation file into edgeDF file
annotated_edgeDF <- merge(edgeDF, annotations, by='row.names')

#Re-create row names and remove redundant columns
rownames(annotated_edgeDF) <- annotated_edgeDF[,1]
annotated_edgeDF <- annotaged_edgeDF[,c(2:30,32,33)]

#Write to file
write.csv(annotated_edgeDF, '../../results/edgeR_allcomparisons_annotated.csv')

#Subset dataframes based on relevant comparisons:
NB.H6 <- DEGs_reordered[,1:6]
NB.D1 <- DEGs_reordered[,c(1,7:11)]
NB.D3 <- DEGs_reordered[,c(1,12:16)]

#Filter based on logFC >= 1, <= -1, p-value <=0.05, FDR <= 0.05
NB.H6_filtered <- filter(NB.H6, NB.H6_logFC >= 1 | NB.H6_logFC <= -1, NB.H6_PValue <=0.05, NB.H6_FDR <=0.05)
NB.D1_filtered <- filter(NB.D1, NB.D1_logFC >= 1 | NB.D1_logFC <= -1, NB.D1_PValue <=0.05, NB.D1_FDR <=0.05)
NB.D3_filtered <- filter(NB.D3, NB.D3_logFC >= 1 | NB.D3_logFC <= -1, NB.D3_PValue <=0.05, NB.D3_FDR <=0.05)

#Write CSVs
write.csv(NB.H6_filtered, 'NB.H6_filtered.csv')
write.csv(NB.D1_filtered, 'NB.D1_filtered.csv')
write.csv(NB.D3_filtered, 'NB.D3_filtered.csv')

#Grab gene names from DEG lists
NB.H6.genes <- as.list(NB.H6_filtered$Gene_names)
NB.D1.genes <- as.list(NB.D1_filtered$Gene_names)
NB.D3.genes <- as.list(NB.D3_filtered$Gene_names)

#Plot Venn diagram with `Venn` function from `gplots` package (v3.0.4)
png('../figures/DEGVenn.png', width=8, height=8, res=300, units='in')
venn(list(NB.6h = NB.H6.genes, NB.d1 = NB.D1.genes, NB.d3 = NB.D3.genes))
dev.off()

```

#### Functional Enrichment Analysis

The DEGs for all comparisons were put into one file and uploaded to [DAVID 6.8](https://david.ncifcrf.gov/home.jsp) and [Metascape](https://metascape.org/gp/index.html#/main/step1) for functional enrichment.

#### Sub-Pathway Analysis and Heatmaps

I created two files with the names of the genes I extracted from the DAVID analysis for two pathways: phagosome formation and phagocytosis.

I then merged them with the DEG file:
```R
library(lattice)
library(gplots)
library("RColorBrewer")

#Set palette
hmcol <- colorRampPalette(brewer.pal(9, "RdBu"))(100)

#Read in files
phagosome_genes <- read.csv('phagosomegenes.csv', header=F)
phagocytosis_genes <- read.csv('phagocytosisgenes.csv', header=F)

phagosome_counts <- merge(phagosome_genes, DEGs_reordered, by.x='V1', by.y='row.names')
phagocytosis_counts <- merge(phagocytosis_genes, DEGs_reordered, by.x='V1', by.y='row.names')

#Write phagocytosis heatmap data
phagocytosis_counts <- read.csv('phagocytosis_counts.csv')

#Data cleanup

y <- as.matrix(phagocytosis_counts)

y <- y[1:9,]

rownames(y) <- y[,3]

y <- y[,c(4,9,14)]

colnames(y) <- c('NB.H6', 'NB.D1', 'NB.D3')

outprefix <- 'phagocytosis'

png("Phagocytosis_heatmap.png", width=8, height=8, res=300, units='in')
levelplot(t(y), height=0.3, col.regions=rev(hmcol), main="Phagocytosis Gene Heatmap", colorkey=list(space="top"), xlab="", ylab="", pretty=TRUE, width=0.5, cexRow=0.1, cexCol=0.1, aspect=2.5)
dev.off()

#Write phagosome heatmap data
phagosome_counts <- read.csv('phagosome_counts.csv')

#Data cleanup

z <- as.matrix(phagosome_counts)

z <- z[c(1:3,5:9),]

rownames(z) <- z[,3]

z <- z[,c(4,9,14)]

colnames(z) <- c('NB.H6', 'NB.D1', 'NB.D3')

png("Phagosome_heatmap.png", width=8, height=8, res=300, units='in')
levelplot(t(z), height=0.3, col.regions=rev(hmcol), main="Phagosome Gene Heatmap", colorkey=list(space="top"), xlab="", ylab="", pretty=TRUE, width=0.5, cexRow=0.1, cexCol=0.1, aspect=2.5)
dev.off()
```

___

### Results

#### Read Quality

The FASTQ quality score was good, with Phred scores above 28 for all of my files, with some over 32 for all reads.

#### Counting and Aligning Statistics

On average, 85% of each `bam` file was aligned and counted by `featureCounts`.

| __Sample__           | % reads aligned |
|------------------|:---------------:|
| SRR121484_0.bam  |       85.8      |
| SRR121484_10.bam |       85.5      |
| SRR121484_11.bam |       85.1      |
| SRR121484_12.bam |       85.7      |
| SRR121484_13.bam |       85.5      |
| SRR121484_14.bam |       85.6      |
| SRR121484_15.bam |       85.9      |
| SRR121484_1.bam  |       86.1      |
| SRR121484_2.bam  |       86.4      |
| SRR121484_3.bam  |       86.2      |
| SRR121484_4.bam  |       85.4      |
| SRR121484_5.bam  |       86.6      |
| SRR121484_6.bam  |       84.8      |
| SRR121484_7.bam  |       83.9      |
| SRR121484_8.bam  |       84.5      |
| SRR121484_9.bam  |       86.1      |
| Average          |       85.5      |

#### PCA Analysis
VSD and rlog are two algorithms used to calculate variance, and they can each offer different results. The analysis is also presented grouped together by experimental condition or by sample.

__Group VSD PCA__
![GroupVSDPCA](https://github.com/swd12012/ee282/blob/finalProject/project/figures/group_vsd_PCA.png)

__Sample VSD PCA__
![SampleVSDPCA](https://github.com/swd12012/ee282/blob/finalProject/project/figures/sample_vsd_PCA.png)

__Group R-log PCA__
![GroupRLogPCA](https://github.com/swd12012/ee282/blob/finalProject/project/figures/group_rlog_PCA.png)

__Sample R-log PCA__
![SampleRLogPCA](https://github.com/swd12012/ee282/blob/finalProject/project/figures/sample_rlog_PCA.png)


#### DEG Analysis

![DEG Venn Diagram](https://github.com/swd12012/ee282/blob/finalProject/project/figures/DEGVenn.png)

Looking at the Venn diagram of DEGs, 40 DEGs are shared across all four time-points. The largest changes occured 6 hours after photobleaching, with fewer DEGs at 1 day post-bleach and the fewest at 3 days post bleach. This corresponds to the analysis done in the paper, which showed a return to "normal" 3 days post-bleach. However, their peak in DEGs occurred at 1 day, not 6 hours.

#### Heatmaps

![Enrichment Heatmap](https://github.com/swd12012/ee282/blob/finalProject/project/figures/HeatmapSelectedGOParent.png)

![Phagocytosis Heatmap](https://github.com/swd12012/ee282/blob/finalProject/project/figures/Phagocytosis_heatmap.png)

![Phagosome Heatmap](https://github.com/swd12012/ee282/blob/finalProject/project/figures/Phagosome_heatmap.png)
___

### Discussion

I demonstrated that I can take raw Illumina reads and take them through an RNAseq pipeline to find DEGs. My pipeline did vary from my proposed analysis, though I believe that I was still able to successfully analyze my data.

After running FASTQC on my files, aligning, and counting, I ended up with an average of 85.5% reads aligned and counted. My Phred scores were over 28 for all reads, though if I trimmed the reads with a higher stringency, I could have perhaps had better alignment.

Looking at my PCA plots, by both VSD and rlog analysis, the 6-hour-post-bleach group had the most variance. This could be a result of heterogeneity in the photobleaching protocol, but it means that caution must be used when this timepoint is analyzed. Subsequent analysis can also exclude these samples and see what difference that could potentially make. Sample 13 in the sample VSD analysis (dark green) also was a potential outlier. One major caveat of RPE and retina preparations is that it is impossible to cleanly separate RPE and retina during a dissection and they can contaminate each other's tissue preps, and so some samples may include more retina than others.

Notably, the analysis in the paper showed the most DEGs at 1 day post-bleach, rather than at 6 hours, and found far less DEGs than I did by a factor of 3 or 4. Discrepancies can be due to the different analysis pipelines and stringencies for cutoffs, etc, as well as which groups were compared.

I generated heatmaps for DEGs that were chosen through gene ontology analysis that related to phagocytosis and the phagosome. The gene _Mir6516_, also known as MER proto-oncogene tyrosine kinase (MerTK) was identified as significantly upregulated at 1 day post-bleach. This aligns with numerous studies in the field that identify this as a critical receptor in mediating RPE phagocytosis, suggesting that the RPE increases its phagocytosis of photoreceptor segments following this insult. As well, since RPE phagocytosis is diurnal, its upregulation at 1 day post bleach, rather than 6 hours post bleach, may be a result of its intrinsic regulation, though the signals that drive phagocytosis are yet to be completely elucidated.

Another interesting gene that was identified in this analysis was _Gm24170_, or the tubby bipartite transcription factor gene. It is a transcription factor with unclear functions, and may play a role in modulating the RPE response to perturbations in homeostasis. 

I had some trouble initially interpreting the data, as the gene names given to me by the ENSEMBL annotation were not listed as human gene names. For interpretation of these gene names I referred to the gene descriptions (found [here](https://github.com/swd12012/ee282/blob/finalProject/project/results/phagocytosis_counts.csv) and [here](https://github.com/swd12012/ee282/blob/finalProject/project/results/phagosome_counts.csv)) I am unsure if an error happened during the annotation where the gene name, ENSEMBL ID, and gene description got scrambled, or if these are the 'correct' ENSEMBL gene names. Subsequent analysis can reannotate the genes with the human orthologues.

Future steps for analysis of this dataset may include determining if any targets of miR-204/211 are found as DEGs within this dataset. I can also cluster the data to see if there are subsets of RPE that respond differently from one another and focus on differential expression between groups. 

Further, more careful analysis of this dataset will be required to determine if this model will be applicable to the specific project I have in mind, but the lessons I learned from this project will aid me in designing pipelines for my future projects, and I have identified genes that warrant a closer look. 

___

#### Works Cited

Luu, J. et al. 'Epigenetic hallmarks of age-related macular degeneration are recapitulated in a photosensitive mouse model.' _Human Molecular Genetics_. 29(15):2611-2624. 1 Aug 2020. (https://doi.org/10.1093/hmg/ddaa158).