## Final Project Report
#### Sam Du
#### EE282 Fall 2020

___

### Methods

Analysis was done in interactive mode. `srun -A ecoevo282 --pty --x11 bash -i`

##### Reference Genome

The _Mus musculus_ reference genome was downloaded from ENSEMBL. I used the GRCm38 release of the top level genome. I also downloaded the GTF for this genome. Checksums were verified against the values generated by `sum` and the ENSEMBL provided checksums.

```bash
wget ftp://ftp.ensembl.org/pub/release-102/fasta/mus_musculus/dna/Mus_musculus.GRCm38.dna.toplevel.fa.gz
wget ftp://ftp.ensembl.org/pub/release-102/gtf/mus_musculus/Mus_musculus.GRCm38.102.gtf.gz 
```

I indexed the genome with BWA index (v0.7.8-r455).

##### Data Acquitisition

Data was downloaded through the European Nucleotide Archive. FTP links for runs SRR12148400 through SRR12148415 were added into a file `download.txt` located in `/data/rawdata/`. Files were batch downloaded with wget with the following command:

```bash
wget -i download.txt
```

##### Data Pre-Processing

FASTQC was used to analyzed read quality. The shell scripts `fastqc.sh` and `fastqc_dir.sh` in the directory `/scripts/` were used to run FASTQC on all the `*.gz` files in my specified directory and output the FastQC files in the same directory:

```bash
sh /data/homezvol2/swdu/ee282/project/scripts/fastqc_dir.sh /data/homezvol2/swdu/ee282/project/data/rawdata/ /data/homezvol2/swdu/ee282/project/data/rawdata/
```

##### Create GTF feature database

I used the R package 'GenomicFeatures' (v1.42.1) to create a GTF SQLite database for use in downstream analysis:

```bash
module load R/4.0.2
```

```R
library(GenomicFeatures)

txdb <- makeTxDbFromGFF(file='Mus_musculus.GRCm38.102.gtf.gz', format='gtf', dataSource='ENSEMBL', organism = 'Mus musculus')

saveDb(txdb, file='Mus_musculus.sqlite')
```