## Final Project Report
#### Sam Du
#### EE282 Fall 2020

___

### Introduction

The retinal pigment epithelium (RPE) is a monolayer of epithelial cells that underlies the retina in the eye. It serves multiple functions, including delivering nutrients photoreceptors, phagocytosing photoreceptor outer segments (POS), and regeneration of ll-cis-retinal for the visual cycle. Recent work has implicated the micro-RNA family miR-204/211 in normal RPE physiology, and that this microRNA family is both (1) light regulated and (2) involved in the regulation of dinural phagocytosis. Phagocytosis is important for the maintenance of health of photoreceptors, shown by both the genetic evidence that disruption of phagocytosis results in retinal degeneration and by structural evidence which shows that the apical RPE membrane is in close physical contact with photoreceptor outer segments. There is a peak in phagocytosis of the distal POS one hour after the switch from dark to light, and as the RPE is a post-mitotic cell type, tight regulation of phagocytosis is critical for maintaing the health of the RPE. However, much is still unknown about the molecular mechanism of phagocytosis, and more work needs to be done to elucidate the pathways through which the RPE maintains retinal health.

Here in this project, I analyze a bulk RNAseq dataset from a recently published paper that models photoreceptor death in a photosensitive mouse model ([Luu et al. 2020, _Human Molecular Genetics_](https://academic.oup.com/hmg/article/29/15/2611/5874042)). In this model, mice that are double knockouts for the photoreceptor proteins _Abca4_ and _Rhd8_ exhibit rapid photoreceptor degeneration upon bright light exposure. As the RPE is vital for maintaining the health of photoreceptors under physiologic conditions, I hypothesize that the RPE may play a pivotal role when dealing with a stressor such as light induced degeneration.

The double-knockout mice (DKO) were compared at four different time points: non-bleached (NB), 6 hours, 1 day, and 3 days post-bleach (6h, 1d, and 3d) respectively. After sacrifice, the eye was enucleated and an eyecup was created, at which point RPE was isolated from the eyecup and prepared for library preparation. Four mice were analyzed at each timepoint.

### Methods

##### Reference Genome

The _Mus musculus_ reference genome was downloaded from ENSEMBL. I used the GRCm38 release of the top level genome. I also downloaded the GTF for this genome. Checksums were verified against the values generated by `sum` and the ENSEMBL provided checksums.

```bash
wget ftp://ftp.ensembl.org/pub/release-102/fasta/mus_musculus/dna/Mus_musculus.GRCm38.dna.toplevel.fa.gz
wget ftp://ftp.ensembl.org/pub/release-102/gtf/mus_musculus/Mus_musculus.GRCm38.102.gtf.gz 
```

I indexed the genome with BWA index (v0.7.8-r455).

I also downloaded the ENSEMBL gene annotations from Biomart for GRCm38 that reports gene name, type, description, and ENSEMBL ID and created a file named `ENSEMBL_annotations.txt` in my `references` folder.

##### Data Acquisition

Data was downloaded through the European Nucleotide Archive. FTP links for runs SRR12148400 through SRR12148415 were added into a file `download.txt` located in `/data/rawdata/`. Files were batch downloaded with wget with the following command:

```bash
wget -i download.txt
```

##### Data Pre-Processing

FASTQC was used to analyze read quality. The shell scripts [`fastqc.sh`](https://github.com/swd12012/ee282/blob/finalProject/project/scripts/fastqc.sub) and [`fastqc_dir.sh`](https://github.com/swd12012/ee282/blob/finalProject/project/scripts/fastqc_dir.sh) in the directory `/scripts/` were used to run FASTQC on all the `*.gz` files in my specified directory and output the FastQC files in the same directory:

```bash
sh /data/homezvol2/swdu/ee282/project/scripts/fastqc_dir.sh /data/homezvol2/swdu/ee282/project/data/rawdata/ /data/homezvol2/swdu/ee282/project/data/rawdata/
```

The FASTQ quality score was good, with Phred scores above 28 for all of my files, with some over 32 for all reads.

I created a [targets file](https://github.com/swd12012/ee282/blob/finalProject/project/data/processed/PCA_sample_vsd.pdf) which annotated my .bam files with sample name, group, condition, etc.

##### Data Alignment

Data was aligned with `bwa mem` and submitted to SLURM as a batch job with the two following scripts [1](https://github.com/swd12012/ee282/blob/finalProject/project/scripts/bwamem.sub) [2](https://github.com/swd12012/ee282/blob/finalProject/project/scripts/bwamem10-15.sub).

##### Read counting

I used the function `featureCounts` from the package Subread (v2.0.1) to perform my counts.

```bash
featureCounts -T 16 \
  -a /data/class/ecoevo282/swdu/ee282/project/reference/Mus_musculus.GRCm38.102.gtf.gz \
  -o /data/class/ecoevo282/swdu/ee282/project/data/processed/readCounts.txt \
  /data/class/ecoevo282/swdu/ee282/project/data/processed/*.bam
  ```

On average, 85% of each bam file was aligned and counted by `featureCounts`.

| __Sample__           | % reads aligned |
|------------------|:---------------:|
| SRR121484_0.bam  |       85.8      |
| SRR121484_10.bam |       85.5      |
| SRR121484_11.bam |       85.1      |
| SRR121484_12.bam |       85.7      |
| SRR121484_13.bam |       85.5      |
| SRR121484_14.bam |       85.6      |
| SRR121484_15.bam |       85.9      |
| SRR121484_1.bam  |       86.1      |
| SRR121484_2.bam  |       86.4      |
| SRR121484_3.bam  |       86.2      |
| SRR121484_4.bam  |       85.4      |
| SRR121484_5.bam  |       86.6      |
| SRR121484_6.bam  |       84.8      |
| SRR121484_7.bam  |       83.9      |
| SRR121484_8.bam  |       84.5      |
| SRR121484_9.bam  |       86.1      |

##### Data Visualization and PCA plotting

I used DESeq2 (v) to create group- and sample-wise PCA plotting by variance stablizing transformation. This will help analyse the variance of each sample and each group and point to potential outliers that may need to be removed in following steps.

First, I cut out the columns that contain only the samples by running the following:

```bash
cat readcounts.txt \
| cut -f7-22 \
> readcounts_samplesonly.txt
```

I then created PCA plots for the dataset in R in the terminal:
```R
library(DESeq2)

#Read in count file as a dataframe
count_dataframe <- as.matrix(read.delim('readcounts_samplesonly.txt'), header=TRUE)

#Reorder count file by column
count_dataframe <- count_dataframe[,c(1,8:16, 2:7)]

#Read in targets file for sample annotation
targets <- read.delim('../targets', comment.char='#', header=TRUE, sep='\t')

#Create dataframes for PCA analysis, one based on sample and one based on group
colDataGroup <- data.frame(row.names=targets$SampleLong, conditions=targets$Factor)
colDataSample <- data.frame(row.names=targets$SampleLong, conditions=targets$SampleLong)

#Create DESeq dataset
sample_dataset <- DESeqDataSetFromMatrix(countData=count_dataframe, colData=colDataSample, design = ~ conditions)
group_dataset <- DESeqDataSetFromMatrix(countData=count_dataframe, colData=colDataGroup, design = ~ conditions)

#Apply variance stabilizing transformation
sample_vsd <- varianceStabilizingTransformation(sample_dataset)
group_vsd <- varianceStabilizingTransformation(group_dataset)

#Apply regularized log transformation
sample_rlog <- rlog(sample_dataset)
group_rlog <- rlog(group_dataset)

#Plot graphs
png('../../figures/sample_vsd_PCA.png', width=8, height=8, res=300, units='in')
plotPCA(sample_vsd, 'conditions')
dev.off()

png('../../figures/group_vsd_PCA.png', width=8, height=8, res=300, units='in')
plotPCA(group_vsd, 'conditions')
dev.off()

png('../../figures/sample_rlog_PCA.png', width=8, height=8, res=300, units='in')
plotPCA(sample_rlog, 'conditions')
dev.off()

png('../../figures/group_rlog_PCA.png', width=8, height=8, res=300, units='in')
plotPCA(group_rlog, 'conditions')
dev.off()
```

__Group VSD PCA__
![GroupVSDPCA](https://github.com/swd12012/ee282/blob/finalProject/project/figures/group_vsd_PCA.png)

__Sample VSD PCA__
![SampleVSDPCA](https://github.com/swd12012/ee282/blob/finalProject/project/figures/sample_vsd_PCA.png)

__Group R-log PCA__
![GroupRLogPCA](https://github.com/swd12012/ee282/blob/finalProject/project/figures/group_rlog_PCA.png)

__Sample R-log PCA__
![SampleRLogPCA](https://github.com/swd12012/ee282/blob/finalProject/project/figures/sample_rlog_PCA.png)

##### Differentially Expressed Gene (DEG) Analysis

DEG analysis was conducted with `edgeR`(v3.30.3) and `systemPipeR` (v1.24.2); both are R packages downloaded via Bioconductor. R version was 3.6.2 to accomodate `systemPipeR`.

```R
library(edgeR)
library(systemPipeR)

#Read in raw counts file and targets file

targets <- as.data.frame(read.delim('../targets.txt', header=T, comment.char='#'))
countDF <- as.data.frame(read.table('readcounts_sampleENSEMBL.txt', header=T, row.names=1))

#Define comparisons to be made, which are stated in the first line of the targets file
cmp <- readComp(file='../targets.txt', format='matrix', delim='-')

#Run edgeR
edgeDF <- run_edgeR(countDF=countDF, targets=targets, cmp=cmp[[1]], independent=TRUE, mdsplot="")

#Read in ENSEMBL annotations and remove first column (redundant information)
annotations <- read.delim('../../reference/ENSEMBL_annotations.csv', header=T, sep=',', row.names=1)
annotations <- annotations[,3-5]

#Merge annotation file into edgeDF file
annotated_edgeDF <- merge(edgeDF, annotations, by='row.names')

#Re-create row names and remove redunant columns
rownames(annotated_edgeDF) <- annotated_edgeDF[,1]
annotated_edgeDF <- annotaged_edgeDF[,c(2:30,32,33)]

#Write to file
write.csv(annotated_edgeDF, '../../results/edgeR_allcomparisons.csv')

```
